packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -477,8 +477,34 @@ export function appendChildToContainer(
     parentNode = (container.parentNode: any);
     parentNode.insertBefore(child, container);
   } else {
-    parentNode = container;
-    parentNode.appendChild(child);
+    if (child.tagName === 'HTML' && container.nodeType === DOCUMENT_NODE) {
+      child = ((child: any): Instance);
+      const doc = ((container: any): Document);
+      if (!doc.documentElement) {
+        doc.write('<html><head></head><body></body></html>');
+      }
+      const documentElement = ((doc.documentElement: any): Instance);
+
+      parentNode = documentElement;
+      copyAttributes(documentElement, child);
+
+      const newHead = child.getElementsByTagName('head')[0];
+      if (newHead) {
+        transferHead(doc, newHead);
```

Comment:
I thought you said head and body works in the browsers we support. Is this just preemptive just in case? Should we get rid of it and wait and see if it comes up?

Reply:
The reason I'm transferring it is so that we preserve the initial head/body elements if they exist instead of constructing new ones. 

Also to clarify I meant that replacing the inner HTML of the document element rather than document does indeed work. I can go with that approach instead if we don't care about preserving the initial head/body. I think there was a test somewhere that was expecting them to be the same I can double check.